https://www.kaggle.com/datasets/ealaxi/paysim1 
✔️ Labeled (isFraud column)
✔️ Transaction-style banking data (TRANSFER, CASH_OUT, etc.)
✔️ Widely used in research (looks legit in reports)
✔️ Easier to model than IEEE-CIS
✔️ Good balance between realism and usability

In [26]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [27]:
ps = pd.read_csv(r"C:\Users\louis\Downloads\R&D PROJECTS\data\PS_20174392719_1491204439457_log.csv")
ps.head()

step      type    amount     nameOrig  oldbalanceOrg  newbalanceOrig  \
0     1   PAYMENT   9839.64  C1231006815       170136.0       160296.36   
1     1   PAYMENT   1864.28  C1666544295        21249.0        19384.72   
2     1  TRANSFER    181.00  C1305486145          181.0            0.00   
3     1  CASH_OUT    181.00   C840083671          181.0            0.00   
4     1   PAYMENT  11668.14  C2048537720        41554.0        29885.86   

      nameDest  oldbalanceDest  newbalanceDest  isFraud  isFlaggedFraud  
0  M1979787155             0.0             0.0        0               0  
1  M2044282225             0.0             0.0        0               0  
2   C553264065             0.0             0.0        1               0  
3    C38997010         21182.0             0.0        1               0  
4  M1230701703             0.0             0.0        0               0

In [28]:
#Set up features + label

from sklearn.model_selection import train_test_split

# Drop IDs (not useful for ML)
X = ps.drop(["isFraud", "isFlaggedFraud", "nameOrig", "nameDest"], axis=1)
y = ps["isFraud"]

# One-hot encode transaction type
X = pd.get_dummies(X, columns=["type"], drop_first=True)

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

In [29]:
#Train random forest
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(
    n_estimators=100,
    random_state=42,
    n_jobs=-1
)

rf.fit(X_train, y_train)

RandomForestClassifier(n_jobs=-1, random_state=42)

In [31]:
#Classification report
from sklearn.metrics import classification_report, roc_auc_score

y_pred = rf.predict(X_test)
y_proba = rf.predict_proba(X_test)[:, 1]

print(classification_report(y_test, y_pred))
print("ROC-AUC:", roc_auc_score(y_test, y_proba))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00   1270881
           1       0.98      0.79      0.87      1643

    accuracy                           1.00   1272524
   macro avg       0.99      0.89      0.94   1272524
weighted avg       1.00      1.00      1.00   1272524

ROC-AUC: 0.9949965048447854


On the PaySim synthetic banking transaction dataset, the Random Forest achieved 98% precision and 79% recall for fraud detection, with ROC-AUC of 0.995. This demonstrates strong fraud detection performance in a banking-style transaction setting, complementing the credit card fraud benchmark.

Performance is strong on PaySim, noting that the dataset is synthetic and therefore may be easier than real-world fraud scenarios.